internal/venice-test-common/src/integrationTest/java/com/linkedin/venice/endToEnd/TestBatch.java

```diff
@@ -650,6 +653,41 @@ public void testKafkaInputAAStore() throws Exception {
     testRepush(storeName, validator);
   }
 
+  @Test(timeOut = TEST_TIMEOUT)
+  public void testKafkaInputBatchJobSnapshots() throws Exception {
+
+    VPJValidator validator = (avroClient, vsonClient, metricsRepository) -> {
+      for (int i = 1; i <= 100; i++) {
+        Assert.assertEquals(avroClient.get(Integer.toString(i)).get().toString(), "test_name_" + i);
+      }
+    };
+
+    String storeName = testBatchStore(
+        inputDir -> new KeyAndValueSchemas(writeSimpleAvroFileWithStringToStringSchema(inputDir)),
+        properties -> {},
+        validator);
+
+    // Re-push with Kafka Input
+    testRepush(storeName, validator);
```

Comment:
It's good to test Kafka Input but can we have a test that's for regular batch too? You can use a boolean like other test methods to control which type of push jobs you want to kick off.